### DISCUSSION TITLE
AC Charge on SNA5000

### DISCUSSION THREAD

**Discussion by Rid00** [celsworth/lxp-bridge#182] [Q&A] [answered]
Hi Chris. Thanks for the great work. I have an SNA5000. I've set the AC Charge times to 00:00-23:59 as requested. Toggling AC Charge on and off does nothing. Lower and upper limits have been set respectively. Not sure if there's anything else I need to do? When checking register 21 with mqtt. AC Charge does indeed change with the toggle. 

Does the standard list of charge options on the invert app matter? Like "According to SOC" and "According to time" changing these doesn't help either. 

Could I also request a short description of some of the published entities? Like the system charge and discharge rates?

**Comment by celsworth** [comment]
Not sure I'm afraid, is the honest answer. This SNA5000 is quite a bit newer than my inverter so there may be quite a lot of changes made I'm not aware of. "According to time" seems like the obvious setting for that, otherwise I guess it will only AC Charge below a certain SOC threshold (and not sure which register that would be set in).

Presumably you can get AC Charge going by just using various buttons on the Luxpower portal? Perhaps a debugging step would be a way to dump all registers before enabling it via their portal, then again afterwards, so we can see what changed. (This is already possible by just scripting up some appropriate mosquitto_pub commands to read all holdings but it could probably be a one-step operation and log the results)..

**Comment by celsworth** [comment, marked as answer]
Since I last posted, https://github.com/celsworth/lxp-bridge/discussions/182 now has quite a lot more information on how to get this going. Lets move any further discussion over there to keep it centralised
